fetch.bs

```diff
@@ -6710,6 +6748,471 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+    <a>share deferred-fetch quota</a>, then decrement <var>quota</var> by
+    <var>otherNavigable</var>'s <a>navigable container</a>'s
+    <a>subframe reserved deferred-fetch quota</a>.
+
+    <p class=note>This is a cross-origin/cross-agent subframe, so we check if it has reserved some
+    quota which was delegated via {{PermissionsPolicy}}.
+  </ol>
+ <li><p>If <var>quota</var> is less than zero, than return zero.
+
+ <li><p>If <var>quota</var> is less than <var>quotaForOrigin</var>, then
+ return <var>quota</var>.
+
+ <li><p>Return <var>quotaForOrigin</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To <dfn export>reserve deferred-fetch quota</dfn> for a <a>navigable container</a>
+<var>container</var> given an <a for=/>origin</a> <var>originToNavigateTo</var>:
+
+<p class=note>This is called when <var>container</var> and the document that initiated the
+navigation (the "source document") <a>share deferred-fetch quota</a>. It potentially
+reserves either 64kb or 8kb of quota for the frame, if it doesn't <a>share deferred-fetch quota</a>
+with its parent and the permissions policy allow. It is not observable to the cotnainer document
+whether the reserved quota was used in practice.
+This algorithm assumes that the container's document might delegate quota to the navigated frame,
+and the reserved quota would only apply in that case, and would be ignored if it ends up being
+shared.
+
+<ol>
+ <li><p>If the <a data-lt="define an inherited policy for feature in container">inherited policy</a>
+ for {{PermissionsPolicy/"deferred-fetch"}}, <var>container</var> and <var>originToNavigateTo</var>
+ is <code>Enabled</code>, and the <a>available deferred-fetch quota</a> for
+ <var>container</var>'s <a>container document</a> is equal or greater than
+ <a>optional subframe deferred-fetch quota</a>, then set <var>container</var>'s
+ <a>reserved deferred-fetch quota</a> to <a>optional subframe deferred-fetch quota</a>.
+
+ <li>
+  <p>Otherwise, if the
+  <a data-lt="define an inherited policy for feature in container">inherited policy</a> for
+  {{PermissionsPolicy/"deferred-fetch-minimal"}}, <var>container</var> and
+  <var>originToNavigateTo</var> is <code>Enabled</code>, and <var>container</var>'s
+  <a>node document</a> and <var>container</var>'s <a>node navigable</a>'s
+  <a for=navigable>top-level traversable</a>'s <a>active document</a>
+  <a>share deferred-fetch quota</a>, then:
+  <ol>
+   <li><p>Let <var>delegatedQuota</var> be <a>deferred-fetch delegated quota</a>.
+
+   <li>
+    <p><a for=list>For each</a> <var>navigable</var> of <var>container</var>'s
+    <a>node navigable</a>'s <a for=navigable>top-level traversable</a>'s
+    <a>inclusive descendant navigables</a> who meets all of the following conditions:
+
+    <ul class=brief>
+     <li><p><var>navigable</var> is not <var>container</var>'s <a>content navigable</a>.
+
+     <li><p><var>navigable</var>'s <a>active document</a> and <var>navigable</var>'s
```

Comment:
```
navigable’s active document and navigable’s top-level traversable’s active document do not share deferred-fetch quota.
```

I am still confused about `navigable's active document` here.

```
root (a.com) -> frame_1 (a.com)
             -> frame_2 (b.com)
```

When `navigable` = `frame_2`, is its `active document` root? So does this step just check whether (root, root) share the same quota?

Reply:
A navigable has an `active document` (internal) and its `container's node document` (its parent).
So when `navigable = frame_2`, its active document is frame_2 and its container's node document is root.